src/Packetery/Module/HookHandler.php

```diff
@@ -60,8 +60,15 @@ public function register(): void {
 	 * @return void
 	 */
 	public function updateOrder( $wcOrderId ): void {
-		$wcOrder = $this->orderRepository->getWcOrderById( (int) $wcOrderId );
+		static $hasBeenRun;
+
+		if ( isset( $hasBeenRun ) && true === $hasBeenRun ) {
+			return;
+		}
+
+		$hasBeenRun = true;
```

Comment:
Nemělo by se nastavit až na úplném konci, tedy až po prvním smazání objednávky ? Že se provede několikrát getWcOrderById() nám nevadí, to je cachováno na straně WC.   Jsme si jisti, že už při prvním spuštění hooku vrací has_shipping_metod změněné údaje ?  Nebude nastavení $hasBeenRun bezpečnější nastavit až na konci ?